### DISCUSSION TITLE
Compile with: --with-ecmult-window=24 --with-ecmult-gen-precision=8

### DISCUSSION THREAD

**Discussion by rs78GT** [bitcoin-core/secp256k1#1429] [Q&A] [answered]
Hi,

I've tried to compile it with those options above (gcc12), to see the performance difference, but it looks like
96G ram is not enough. Probably just with a few less, or only one core I can make it, but I would have
two questions:

-       Is there any speed-up, or performace benefit on [secp256k1_ec_pubkey_tweak_add] function using with that large table
        against the basic size (15, I guess..) setting? This would be the only important function.

-       If someone already compiled it with success.. How much memory is enough? Maybe clang less hungry.. I don't know.
        The 24/8 seems to be the "fastest" settings (according to the config), but anyway.. can I go bigger? :)

Thanks in advance!

**Comment by jonasnick** [comment, marked as answer]
Hi @rs78GT,

compiling with those two options with gcc 12.3 worked for me on a machine with 64G of memory.

The `with-ecmult-window` option affects the function you're interested in (`secp256k1_ec_pubkey_tweak_add`); the `with-ecmult-gen-precision` option does not.

**Reply by rs78GT** [reply]
Hello,
thank you for your quick answer! I did not realize "my question" was moved here (my mistake), but anyway..
This is good news! I try to recompile it again, now on less cores, so I'm hoping for the best.
That is the answer, what I was looking for! :)

Thanks again!

**Reply by jonasnick** [reply]
Ah, yes, I should have noted that I compiled with `make -j 1`.

**Reply by rs78GT** [reply]
Probably 10 was a little overkill.. :)
Have a good day!
Zsolt

**Comment by sipa** [comment]
Independently, `--with-ecmult-gen-precision=8` is not necessarily faster than `--with-ecmult-gen-precision=4`. I recommend benchmarking before you try working with any non-default setting.

The same holds for `--with-ecmult-window`, though the negative effects of larger tables are likely less pronounced there.

**Comment by rs78GT** [comment]
Hello sipa,
thanks for your time! The truth is.. I rewrote it with GMP functions. What I really need is the "simple" generator point addition to another point, so I tried to go lower level. Raw C code, nothing WoodooMagic assembly, but it is surprisingly fast! From 40-50 something seconds I got less than 13 seconds, including the array movements (benchmark: 1.000.000 cycles, Intel CherryTrail X8350). Right now, everything is in MPZ, but I try to figure out, how implement with MPN functions.. just push it down, under to the hell! :)
I guess, this is the way.
